sts/signer.go

```diff
@@ -274,11 +274,20 @@ func (s *Signer) SignRequest(req *http.Request) error {
 		}
 
 		var buf bytes.Buffer
+		host := req.URL.Hostname()
+		// Check if the host IP is in IPv6 format. If yes, add the opening and closing square brackets.
+		// If the host IP contains ":", it is either an IPv6 address or the entire host-port input if the port is not valid.
+		// https://github.com/golang/go/blob/master/src/net/url/url.go#L1148
```

Comment:
Nit: should we point to the method name in that file, as line number may change right away

Reply:
Thanks Zach for the review. Yes updated the comment.